src/lava/magma/runtime/runtime.py

```diff
@@ -201,56 +205,97 @@ def start(self, run_condition: AbstractRunCondition):
+                                error_cnt += 1
+                        self._error = True
+                    else:
+                        raise RuntimeError(f"Runtime Received {data}")
+            if self._req_paused:
+                self._req_paused = False
+                self.pause()
+            if self._req_stop:
+                self._req_stop = False
+                self.stop()
+            if self._error:
+                # self.stop()
+                raise RuntimeError(
+                    f"{error_cnt} Exception(s) occurred. See "
+                    f"output above for details.")
+            self._is_running = False
+
+    def _run(self, run_condition: AbstractRunCondition):
         if self._is_started:
             self._is_running = True
             if isinstance(run_condition, RunSteps):
                 self.num_steps = run_condition.num_steps
                 for send_port in self.runtime_to_service:
                     send_port.send(enum_to_np(self.num_steps))
                 if run_condition.blocking:
-                    for recv_port in self.service_to_runtime:
-                        data = recv_port.recv()
-                        if not enum_equal(data, MGMT_RESPONSE.DONE):
-                            if enum_equal(data, MGMT_RESPONSE.ERROR):
-                                # Receive all errors from the ProcessModels
-                                error_cnt = 0
-                                for actors in \
-                                        self._messaging_infrastructure.actors:
-                                    actors.join()
-                                    if actors.exception:
-                                        _, traceback = actors.exception
-                                        print(traceback)
-                                        error_cnt += 1
-
-                                raise RuntimeError(
-                                    f"{error_cnt} Exception(s) occurred. See "
-                                    f"output above for details.")
-                            else:
-                                raise RuntimeError(f"Runtime Received {data}")
-                if run_condition.blocking:
-                    self._is_running = False
+                    self._get_resp_for_run()
             elif isinstance(run_condition, RunContinuous):
-                pass
+                self.num_steps = sys.maxsize
+                for send_port in self.runtime_to_service:
+                    send_port.send(enum_to_np(self.num_steps))
             else:
                 raise ValueError(f"Wrong type of run_condition : "
                                  f"{run_condition.__class__}")
         else:
             print("Runtime not started yet.")
 
     def wait(self):
+        """
+        Waits for RuntimeServices to send Response
+        """
+        self._get_resp_for_run()
+
+    def pause(self):
+        """
+        Pauses a add_ports_for_polling
```

Comment:
"Period"

Reply:
Ok... it seems all these "add_ports_for_polling" all throughout the code base might have been a refactoring/renaming error.